### PR TITLE
Fix typos and Spell check auto integration 

### PR DESCRIPTION
**Pull request by miles-six**
### Description

I fixed many typos and want to  integrate Spell check to workflow so can fix all typos in the futures

### CODE REVIEW

**Comment by jmrossy** [comment]
Thanks @miles-six for the contribution! A few of these are for `receipient`, which is a valid variant of `recipient`. Can you please have that word ignored?

**Comment by miles-six** [comment]
> Thanks @miles-six for the contribution! A few of these are for `receipient`, which is a valid variant of `recipient`. Can you please have that word ignored?

Yes,I have down it and you even can ignore some words by add to them to .codespell/wordlist.txt

**Comment by miles-six** [comment]
@jmrossy  I have added the word 'receipient' to the word.txt file, and it will be ignored in future pull requests.

File: rust/agents/scraper/migration/src/m20230309_000005_create_table_message.rs [hunk 1]
```diff
@@ -135,7 +135,7 @@ impl MigrationTrait for Migration {
                 "origin_tx"."{tx_egp}" AS "origin_tx_effective_gas_price",
                 "origin_tx"."{tx_nonce}" AS "origin_tx_nonce",
                 "origin_tx"."{tx_sender}" AS "origin_tx_sender",
-                "origin_tx"."{tx_receipient}" AS "origin_tx_recipient",
+                "origin_tx"."{tx_recipient}" AS "origin_tx_recipient",
```
Comment (jmrossy) [inline rust/agents/scraper/migration/src/m20230309_000005_create_table_message.rs:138]: We need to please revert the changes to this database migration file or it will break the Hyperlane explorer
